codemp/rd-rend2/glsl/ssao.glsl

```diff
@@ -8,95 +8,92 @@ void main()
 {
 	gl_Position = attr_Position;
 	var_ScreenTex = attr_TexCoord0.xy;
-	//vec2 screenCoords = gl_Position.xy / gl_Position.w;
-	//var_ScreenTex = screenCoords * 0.5 + 0.5;
 }
 
 /*[Fragment]*/
-uniform sampler2D u_ScreenDepthMap;
-uniform vec4 u_ViewInfo; // zfar / znear, zfar
+uniform sampler2D u_ScreenDepthMap; // colormap
+uniform sampler2D u_ScreenImageMap; // lightmap
+uniform vec4 u_ViewInfo; // znear, zfar, 0, 0
+uniform vec2 u_ScreenInfo; // width, height
+
+uniform vec4 u_SSAOSettings; // aocap, strength, aoMultiplier, lightmap
+uniform vec4 u_SSAOSettings2; // noise, aorange, depthTolerance, 0
 
 in vec2 var_ScreenTex;
 
 out vec4 out_Color;
 
-vec2 poissonDisc[9] = vec2[9](
-vec2(-0.7055767, 0.196515),    vec2(0.3524343, -0.7791386),
-vec2(0.2391056, 0.9189604),    vec2(-0.07580382, -0.09224417),
-vec2(0.5784913, -0.002528916), vec2(0.192888, 0.4064181),
-vec2(-0.6335801, -0.5247476),  vec2(-0.5579782, 0.7491854),
-vec2(0.7320465, 0.6317794)
-);
-
-// Input: It uses texture coords as the random number seed.
-// Output: Random number: [0,1), that is between 0.0 and 0.999999... inclusive.
-// Author: Michael Pohoreski
-// Copyright: Copyleft 2012 :-)
-// Source: http://stackoverflow.com/questions/5149544/can-i-generate-a-random-number-inside-a-pixel-shader
-
-float random( const vec2 p )
-{
-  // We need irrationals for pseudo randomness.
-  // Most (all?) known transcendental numbers will (generally) work.
-  const vec2 r = vec2(
-    23.1406926327792690,  // e^pi (Gelfond's constant)
-     2.6651441426902251); // 2^sqrt(2) (Gelfond-Schneider constant)
-  //return fract( cos( mod( 123456789., 1e-7 + 256. * dot(p,r) ) ) );
-  return mod( 123456789., 1e-7 + 256. * dot(p,r) );  
+//
+// AO Shader by Monsterovich :D
+//
+
+vec2 camerarange = vec2(u_ViewInfo.x, u_ViewInfo.y);
+
+float readDepth( in vec2 coord ) {
+	return (2.0 * camerarange.x) / (camerarange.y + camerarange.x - texture2D( u_ScreenDepthMap, coord ).x * (camerarange.y - camerarange.x));	
 }
 
-mat2 randomRotation( const vec2 p )
-{
-	float r = random(p);
-	float sinr = sin(r);
-	float cosr = cos(r);
-	return mat2(cosr, sinr, -sinr, cosr);
+float compareDepths( in float depth1, in float depth2 ) {
+	float diff = sqrt( clamp(1.0-(depth1-depth2) / ( u_SSAOSettings2.y /* aorange */ / (u_ViewInfo.y - u_ViewInfo.x)),0.0,1.0) );
+	float ao = min(u_SSAOSettings.x /* aocap */,max(0.0,depth1-depth2-u_SSAOSettings2.z /* depthTolerance */) * u_SSAOSettings.z /* aoMultiplier */) * diff;
+	return ao;
 }
 
-float getLinearDepth(sampler2D depthMap, const vec2 tex, const float zFarDivZNear)
+vec2 rand(vec2 coord) //generating random noise
```

Comment:
Remove the comment - it's already obvious from the function name that it generates random numbers.

What's not obvious is the range of numbers that the function can output. Does the function output random numbers between 0 and 1? 0 and 0.004? Perhaps a comment can be added to explain that instead

Reply:
Removed.